CMakePresets.json

```diff
@@ -254,7 +254,10 @@
     },
     {
       "name": "ci-clang-release",
-      "inherits": "clang-release"
+      "inherits": "clang-release",
+      "cacheVariables": {
+        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "TRUE"
+      }
```

Comment:
can you please also add this to `ci-clang-tidy-release` and `ci-clang-tidy-debug`? I wonder if that speeds up clang-tidy

Reply:
Why do we want that? Clang-tidy is not the bottleneck in the CI.

Reply:
we disabled part of the clang-tidy build in CI due to the time it takes to run all checks. If the reason it takes so long is indeed the optimization, and if we can solve it using this flag, then we might enable it in the full build in CI again.

Reply:
I seriously doubt it, but let me try.

Reply:
@mgovers  full build in https://github.com/PowerGridModel/power-grid-model/actions/runs/12888364732

Reply:
It is not faster. I will revert the change.